### PR TITLE
[runtime-detection-x86] detect avx and avx2 only if osxsave is true

### LINKED ISSUE #153: AVX detection needs to check if OS supports save/restore
See https://hg.mozilla.org/mozilla-central/file/64bab5cbb9b6/mozglue/build/SSE.cpp#l190 for an example

### PR DESCRIPTION
**Pull request by gnzlbg**
Closes #153.

### CODE REVIEW

File: src/x86/runtime.rs [hunk 1]
```diff
@@ -111,48 +111,56 @@ fn inv_test_bit(v: usize, idx: u32) -> bool {
...
+
+    // ECX[26] detects XSAVE and ECX[27] detects OSXSAVE, that is, whether the
+    // OS is AVX enabled and supports saving the state of the AVX/AVX2 vector
+    // registers on context-switches, see:
+    //
+    // - https://software.intel.com/en-us/blogs/2011/04/14/is-avx-enabled
+    // - https://hg.mozilla.org/mozilla-central/file/64bab5cbb9b6/mozglue/build/SSE.cpp#l190
+    //
+    if inv_test_bit(proc_info_ecx, 26) && inv_test_bit(proc_info_ecx, 27) {
+        if inv_test_bit(proc_info_ecx, 28) { value = set_bit(value, __Feature::avx as u32); }
```
Comment (jrmuizel) [inline src/x86/runtime.rs]: You need to actually use the xgetbv instruction here to confirm that the OS is going to do the right thing.

**Comment by gnzlbg** [comment]
@jrmuizel indeed, I've updated the PR.

@BurntSushi we might want to expose `xgetbv` on x86 "somewhere".

**Comment by petrochenkov** [comment]
>we might want to expose xgetbv on x86 "somewhere"

BTW, since this crate is supposed to cover only SIMD intrinsics (?), are there any plans to cover everything else?
(`xgetbv` in particular is [available](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=xgetbv&expand=5686) as an intrinsic in `immintrin.h`.)

**Comment by gnzlbg** [comment]
@petrochenkov this crate implements `core::vendor` and aims to cover all vendor intrinsics. It is called `stdsimd` because its main current focus are SIMD intrinsics, but it already covers many non-SIMD-related intrinsics.

**Comment by jrmuizel** [comment]
Looks good to me.

**Comment by alexcrichton** [comment]
Looks great, thanks! I think CI is failing though?

**Comment by gnzlbg** [comment]
Yes, I screwed up the assembly syntax, I'll fix it.

**Comment by gnzlbg** [comment]
@alexcrichton this is ready

**Comment by alexcrichton** [comment]
Awesome, thanks!

**Pull request merged**